docs/en/operations/update.md

```diff
@@ -1,10 +1,75 @@
 ---
 slug: /en/operations/update
+sidebar_title: Self-managed Upgrade
+title: Self-managed Upgrade
 ---
 
-# Update
+## ClickHouse upgrade overview
 
-## Self-managed ClickHouse Upgrade
+This document contains:
+- general guidelines
+- a recommended plan
+- specifics for upgrading the binaries on your systems
+
+## General guidelines
+
+These notes should help you with planning, and to understand why we make the recommendations that we do later in the document.
+
+### Upgrade ClickHouse server separately from ClickHouse Keeper or ZooKeeper
+Unless there is a security fix needed for ClickHouse Keeper or Apache ZooKeeper it is not necessary to upgrade Keeper when you upgrade ClickHouse server.  Keeper stability is required during the upgrade process, so complete the ClickHouse server upgrades before considering an upgrade of Keeper.
+
+### Minor version upgrades should be adopted often
+It is highly recommended to always upgrade to the newest minor version as soon as it is released. Minor releases do not have breaking changes but do have important bug fixes (and may have security fixes).
+
+
+### Test experimental features on a separate ClickHouse server running the target version
+
+The compatibility of experimental features can be broken at any moment in any way.  If you are using experimental features, then check the changelogs and consider setting up a separate ClickHouse server with the target version installed and test your use of the experimental features there.
+
+### Downgrades
+If you upgrade and then realize that the new version is not compatible with some feature that you depend on you may be able to downgrade to a recent (less than one year old) version if you have not started to use any of the new features.  Once the new features are used the downgrade will not work.
+
+### Minor version upgrades
```

Comment:
Did you mean "Major version upgrades" here?

Reply:
I am in the process of changing that title to:

### Multiple ClickHouse server versions in a cluster 

and adding in the warnings that you added to the issue.

Reply:
updated in https://github.com/ClickHouse/ClickHouse/pull/44083